### ISSUE TITLE
[ENH]: pcolormesh : values not printed in the navigation toolbar

### ISSUE THREAD

**Issue by patquem** [matplotlib/matplotlib#24406]
### Problem

Hello,
When viewing a 2D map with plt.pcolormesh(), the values ​​corresponding to the current mouse positions are not printed in the navigation toolbar (unlike plt.imshow()).
Is it an oversight, a bug or an incompatibility?
If none of these, could you please add this feature in the future ?
Thanks,
Patrick

**Comment by greglucas** [comment]
pcolormesh cursor data turned out to be really slow for large meshes https://github.com/matplotlib/matplotlib/issues/21917, so it is now disabled by default. To get this capability you can turn on the mouseover property of the returned artist.

```python
mesh = plt.pcolormesh()
mesh.set_mouseover(True)
```

**Comment by patquem** [comment]
Thank you very much @greglucas,
Unfortunately This information cannot be found easily (on my opinion).
Maybe revisit the default choice (mouseover=True by default ?) considering that small basic applications are more common that large calculations ?
Regards,
Patrick

**Comment by greglucas** [comment]
> Unfortunately This information cannot be found easily (on my opinion).

I agree, it is not very well advertised or documented. PRs adding an example/documentation about this would be welcome.

> Maybe revisit the default choice (mouseover=True by default ?)

I thought there was some discussion around whether to set a different default for different size meshes, but I can't find it from a quick search now. I can certainly see an argument that if `mesh.size() < N` where N ~100 (I'm not sure on the actual size where performance degrades) we could enable it by default. But, that may also cause some confusion when someone is making various mesh sizes that straddle that boundary and gets different interactions.

**Comment by jklymak** [comment]
Crossref https://github.com/matplotlib/matplotlib/pull/22693

**Comment by timhoffm** [comment]
Additional note: When deactivated, the mouse over could display a hint like "Set pcolormesh(..., mouseover=True) to show coordinates" - if there is enough space. Shorter messages welcome.
